### PR TITLE
Remove unspecified TRP warnings for projects without ProjectReferences

### PR DESCRIPTION
**Pull request by jbayardo**
When an MSBuild project does not have any ProjectReferences, no warnings due to a missing Target Reference Protocol should be emitted.

### CODE REVIEW

**Comment by smera** [comment]
@cdmihai is added to the review. #Closed

File: Public/Src/FrontEnd/MsBuild/PipConstructor.cs [hunk 1]
```diff
@@ -560,7 +560,7 @@ private bool TryAddMsBuildArguments(ProjectWithPredictions project, PipDataBuild
                     pipDataBuilder.Add(PipDataAtom.FromString($"/t:{target}"));
                 }
             }
-            else
+            else if (project.ProjectReferences.Count > 0)
```
Comment (smera) [inline Public/Src/FrontEnd/MsBuild/PipConstructor.cs]: >if (project.ProjectReferences.Count > 0) [](start = 17, length = 40)

Not sure about this. Shouldn't all projects comply to the project reference protocol in this new world of static graph MSBuild?
I understand that when projects are leaves the protocol is inconsequential, but a project not specifying the protocol sounds fishy enough to me to do something about that (we warn today, but we may decide to error in the future). 
@micodoba@microsoft.com , thoughts?

Comment (jbayardo) [reply Public/Src/FrontEnd/MsBuild/PipConstructor.cs]: Why is it fishy? In CloudBuild, for example, there are many leaf projects that are just moving files around, or strong-signing NuGet dependencies; moreover, there are roughly 40 of these.

In cases like this, warning about missing the TRP for leaves is just adding noise, as there may be many other projects which genuinely need attention. Furthermore, the warnings are misleading: there are no references, so what should the TRP be?

Comment (cdmihai) [reply Public/Src/FrontEnd/MsBuild/PipConstructor.cs]: Is there an option to downgrade the warning to informational logs for leaf nodes?

Comment (jbayardo) [reply Public/Src/FrontEnd/MsBuild/PipConstructor.cs]: I didn't see any way to downgrade the warning, so I just added a new informational event specific for the leaves.

Comment (smera) [reply Public/Src/FrontEnd/MsBuild/PipConstructor.cs]: In an ideal world all SDKs implement the protocol. This protocol specification goes beyond the concrete references a node may have. So even if a node doesn't have any references, the protocol will say what happens if references are added.

But after discussing this with Mihai, he convinced me that in our current not-ideal world, it is ok to let that slide. So I'm good.

---
In reply to: [281338766](https://github.com/Microsoft/BuildXL/pull/256#discussion_r281338766) [](ancestors = 281338766)

**Comment by smera** [comment]
>             EventLevel = Level.Warning,

Talking with Mihai about this, we also agreed that this should probably be turned into an error. Things are almost bound to fail anyway, and it is better if we start with a more strict set of constraints and relax them later than the other way around.
@julian@bayardo.info , maybe you want to do that as port of this PR? Sounds close enough to the changes you are already proposing. 
Otherwise I'll send a separate PR with that change, just let me know.

---
Refers to: Public/Src/FrontEnd/MsBuild/Tracing/Log.cs:228 in 865749b. [](commit_id = 865749b0247731e1588bd6939eaea1e28781d7d4, deletion_comment = False)

**Comment by jbayardo** [comment]
As we spoke, although we definitely should turn it into an error, that is not something we can do right now due to constraints from the CB repo, which is on-going work.

**Pull request merged**